network-dashboard-proxy/src/main.rs

```diff
@@ -0,0 +1,253 @@
+use anyhow::Context;
+use axum::{
+    http::{self, StatusCode},
+    routing::get,
+    Router,
+};
+use clap::Parser;
+use concordium_rust_sdk::{
+    endpoints::QueryError,
+    types::{
+        hashes::{BlockHash, TransactionHash},
+        queries::{BlockInfo, ConsensusInfo},
+        AbsoluteBlockHeight, TransactionStatus,
+    },
+    v2,
+};
+use futures::TryStreamExt;
+use tonic::transport::ClientTlsConfig;
+use tower_http::trace::{DefaultMakeSpan, DefaultOnResponse};
+
+#[derive(clap::Parser, Debug)]
+#[clap(arg_required_else_help(true))]
+#[clap(version, author)]
+struct App {
+    #[clap(
+        long = "node",
+        help = "GRPC V2 interface of the node.",
+        default_value = "http://localhost:20000",
+        env = "NETWORK_DASHBOARD_PROXY_CONCORDIUM_NODE"
+    )]
+    endpoint:        v2::Endpoint,
+    #[clap(
+        long = "listen-address",
+        default_value = "0.0.0.0:8080",
+        help = "Listen address for the server.",
+        env = "NETWORK_DASHBOARD_PROXY_API_LISTEN_ADDRESS"
+    )]
+    listen_address:  std::net::SocketAddr,
+    #[clap(
+        long = "log-level",
+        default_value = "info",
+        help = "Maximum log level.",
+        env = "NETWORK_DASHBOARD_PROXY_LOG_LEVEL"
+    )]
+    log_level:       tracing_subscriber::filter::LevelFilter,
+    #[clap(
+        long = "log-headers",
+        help = "Whether to log headers for requests and responses.",
+        env = "NETWORK_DASHBOARD_PROXY_LOG_HEADERS"
+    )]
+    log_headers:     bool,
+    #[clap(
+        long = "request-timeout",
+        help = "Request timeout in milliseconds.",
+        default_value = "5000",
+        env = "NETWORK_DASHBOARD_PROXY_REQUEST_TIMEOUT"
+    )]
+    request_timeout: u64,
+}
+
+#[derive(Debug, thiserror::Error)]
+enum Error {
+    #[error("Unable to query node: {0:?}")]
+    Query(#[from] QueryError),
+    #[error("Network error: {0:?}")]
+    Network(#[from] v2::Status),
+}
+
+impl axum::response::IntoResponse for Error {
+    fn into_response(self) -> axum::response::Response {
+        let r = match self {
+            Self::Query(err) => {
+                if err.is_not_found() {
+                    (
+                        StatusCode::NOT_FOUND,
+                        axum::Json("Requested value not found.".to_string()),
+                    )
+                } else {
```

Comment:
This case is also catching every RPC status, should these be mapped to bad gateway?

Reply:
I think so. For the queries that are made, since the inputs are parsed so we know that they are well-formed, then either the object isn't found or something is wrong with the node connection or node configuration, where bad_gateway seems an appropriate error.